File: tests/cases/check/layer-dependencies/python/inputs/src/domain/order.py
from cdec_rules import layer

from infrastructure.sql import SqlConnection
from support.logger import Logger


@layer("domain")
class Order:
    """The violation: `domain` allows nothing, and SqlConnection is
    `infrastructure`. The `Logger` reference is invisible - its class carries
    no `@layer` tag, and the rule cannot reason about untagged targets."""

    def __init__(self, conn: SqlConnection, log: Logger) -> None:
        self.conn: SqlConnection = conn
        self.log: Logger = log

File: tests/cases/check/layer-dependencies/csharp/inputs/src/Domain/Order.cs
using CodeConstraints.Rules;

using Infrastructure;
using Support;

namespace Domain;

// The violation: `domain` allows nothing, SqlConnection is `infrastructure`.
// The `Logger` reference is invisible - Logger carries no [Layer] tag.
[Layer("domain")]
public class Order
{
    public SqlConnection Conn { get; }
    public Logger Log { get; }

    public Order(SqlConnection conn, Logger log) { Conn = conn; Log = log; }
}
